pkg/util/secretutil/secret.go

```diff
@@ -84,6 +102,19 @@ func getPrivateKey(data []byte) ([]byte, error) {
 		data = rest
 	}
 
+	// if both der and pemBlock are nil, then it might be a pfx cert
+	if der == nil && pemBlock == nil {
+		key, _, err := pkcs12.Decode(data, "")
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode pkcs12 data, err: %w", err)
+		}
+
+		derKey, err = x509.MarshalPKCS8PrivateKey(key.(*rsa.PrivateKey))
+		if err != nil {
+			return nil, fmt.Errorf("failed to marshal private key, err: %w", err)
+		}
```

Comment:
same comment as above (instead of `MarshalPKCS8PrivateKey`, could we reuse the existing key parse logic?)

Reply:
Same as above. IMO we don't need to parse it.